src/tests/integration/omneo/rewards-definition/delete-reward-definition.test.ts

```diff
@@ -0,0 +1,53 @@
+import { describe, expect, test, afterAll } from 'vitest'
+import { Omneo } from '../../../../omneo'
+import simpleOmneoRequest from '../../../lib/simple-omneo-request'
+import { RewardDefinitionCreateInput } from '../../../../types'
+import { getName, getHandle } from './util'
+
+const omneo = new Omneo({
+  tenant: process.env.OMNEO_TENANT as string,
+  token: process.env.OMNEO_TOKEN as string
+})
+const FAILED_DELETE_REWARDS_DEFINITION_IDS : number[] = []
+
+describe('Reward Definition delete', () => {
+  test('SDK Reward Definition delete', async () => {
+    const payload: RewardDefinitionCreateInput = {
+      name: getName(),
+      handle: getHandle(),
+      value: 10,
+      period: 30,
+      period_type: 'days',
+      type: 'spend'
+    }
+    const response = await simpleOmneoRequest('POST', '/rewards/definitions', payload).catch((err) => {
+      console.error('SDK update reward definition created failed:', err)
+      throw new Error('SDK update reward definition created failed')
+    })
+    await omneo.rewardsDefinition.delete(response.data.id).catch((err) => {
+      console.error(`SDK Reward definition delete failed with id:${response.data.id}`, err)
+      FAILED_DELETE_REWARDS_DEFINITION_IDS.push(response.data.id)
+      throw new Error(`SDK Reward definition delete failed with id:${response.data.id}`)
+    })
+
+    const rewardDefinitionsRes = await omneo.rewardsDefinition.list({
```

Comment:
fetch by id `response.data.id`
`expect` it to receive HTTP 404

Reply:
here is list, it will return empty data, how could it be 404? @oliverarkade

Reply:
you shouldn't use the SDK for this, and also shouldn't use the list endpoint
Fetch the specific reward definition by ID instead. 
And expect it to return a 404